plugin-server/src/cdp/utils.ts

```diff
@@ -158,3 +160,37 @@ export const unGzipObject = async <T extends object>(data: string): Promise<T> =
 
     return JSON.parse(res.toString())
 }
+
+export const prepareLogEntriesForClickhouse = (
+    result: HogFunctionInvocationResult
+): HogFunctionLogEntrySerialized[] => {
+    const preparedLogs: HogFunctionLogEntrySerialized[] = []
+    const logs = result.logs
+    result.logs = [] // Clear it to ensure it isn't passed on anywhere else
+
+    const sortedLogs = logs.sort((a, b) => a.timestamp.toMillis() - b.timestamp.toMillis())
+
+    // Start with a timestamp that is guaranteed to be before the first log entry
+    let timestamp = sortedLogs[0].timestamp.minus(1)
+
+    sortedLogs.forEach((logEntry) => {
+        if (logEntry.timestamp <= timestamp) {
```

Comment:
Probably worth adding the "TRICKY" note here about events being deduped by timestamp, otherwise this looks quite strange

Reply:
true - log got lost in the movements